contractcourt/htlc_success_resolver.go

```diff
@@ -417,31 +417,31 @@ func (h *htlcSuccessResolver) resolveRemoteCommitOutput() (
 		// with the published one.
 	}
 
+	// Register the confirmation notification before broadcasting the sweep
+	// transaction.
+	sweepTXID := h.sweepTx.TxHash()
+	sweepScript := h.sweepTx.TxOut[0].PkScript
+	confNtfn, err := h.Notifier.RegisterConfirmationsNtfn(
```

Comment:
wondering what happens if the publishing of the transaction fails later, will we ever close the notification channel then?

Reply:
yeah we need to cancel it, updated